hs/CalendarResponse.hs

```diff
@@ -0,0 +1,281 @@
+          "ORGANIZER:University of Toronto",
+          "SUMMARY:" ++ code ++ " " ++ sect,
+          "CATEGORIES:EDUCATION", "END:VEVENT"]
+
+-- ** Ordering data
+
+-- | Orders the time fields for each course; which were obtained from the
+-- database
+orderData :: [Time] -> [[[Double]]]
+orderData courseFields = filter notNull (assignDay $ map timeField courseFields)
+    where
+        notNull field = not $ null field
+
+-- | Organizes the data by the day of the week, assigning a position based on
+-- the day in the time field.
+assignDay :: [[Double]] -> [[[Double]]]
+assignDay lst = [monday, tuesday, wednesday, thursday, friday]
+    where
+        monday = filter (\ field -> field !! 0 == 0.0) lst
+        tuesday = filter (\ field -> field !! 0 == 1.0) lst
+        wednesday = filter (\field -> field !! 0 == 2.0) lst
+        thursday = filter (\field -> field !! 0 == 3.0) lst
+        friday = filter (\field -> field !! 0 == 4.0) lst
+
+-- ** Start time
+
+-- | Obtains a list of start times for each course.
+startTime :: [[[Double]]] -> [[String]]
+startTime dataInOrder = map checkTimeStart dataInOrder
+    where
+        checkTimeStart dataByDay = 
+            map getStrTime ([head (sortList dataByDay)] ++
+                        getStartConsecutives (sortList dataByDay))
+        sortList dataDay = sort $ map (!! 1) dataDay
+
+-- | Gets the start times that are not the very first start time. 
+getStartConsecutives :: [Double] -> [Double]
+getStartConsecutives lst = 
+    filter (/= 30.0) ([if lst !! i == lst !! (i + 1) - 0.5
+                       then 30.0
+                       else lst !! (i + 1)| i <- [0 .. l]])
+    where
+        l = length lst - 2
+
+-- ** End time
+
+-- | Obtains a list of end times for each course.
+endTime :: [[[Double]]] -> [[String]]
+endTime dataInOrder = map checkTimeEnd dataInOrder
+    where
+        checkTimeEnd dataByDay =
+            map (getStrTime . (+ 0.5))
+                (getEndConsecutives (sortList dataByDay) ++
+                                     [last (sortList dataByDay)])
+        sortList dataDay = sort $ map (!! 1) dataDay
+
+-- | Gets the end times that are not the very first end time. 
+getEndConsecutives :: [Double] -> [Double]
+getEndConsecutives lst =
+    filter (/= 30.0) ([if lst !! i == lst !! (i + 1) - 0.5
+                       then 30.0
+                       else lst !! i| i <- [0 .. l]])
+    where
+        l = length lst - 2
+
+-- ** Function that works for both start and end times
+
+-- | Creates a string time in the following way HourMinutesSeccondsZ.
+-- For instance, 133500Z corresponds to 1:35 pm.
+getStrTime :: Double -> String
+getStrTime fullTime = if minutes == 0
+                  then hour ++ "0000"
+                  else hour ++ ratio minutes ++ "00"
+    where
+        hours = splitOn "." (show fullTime)
+        hour = hours !! 0
+        minutes = read $ hours !! 1 :: Int
+
+-- | Creates a string for the minutes.
```

Comment:
This function doc doesn't make much sense to me. The name can be improved too. Try to describe precisely what this function does in the name.